packages/services/api/src/modules/organization/providers/organization-manager.ts

```diff
@@ -471,26 +478,72 @@ export class OrganizationManager {
     return result;
   }
 
-  async deleteInvitation(input: { email: string; organizationId: string }) {
+  async deleteInvitation(args: {
+    email: string;
+    organization: GraphQLSchema.OrganizationReferenceInput;
+  }) {
+    const { organizationId } = await this.idTranslator.resolveOrganizationReference({
+      reference: args.organization,
+      onError: () => {
+        this.session.raise('member:modify');
+      },
+    });
     await this.session.assertPerformAction({
       action: 'member:modify',
-      organizationId: input.organizationId,
+      organizationId,
       params: {
-        organizationId: input.organizationId,
+        organizationId,
       },
     });
-    return this.storage.deleteOrganizationInvitationByEmail(input);
+    return this.storage.deleteOrganizationInvitationByEmail({
+      organizationId,
+      email: args.email,
+    });
   }
 
-  async inviteByEmail(input: { email: string; organization: string; role?: string | null }) {
+  async inviteByEmail(input: {
+    organization: GraphQLSchema.OrganizationReferenceInput;
+    email: string;
+    role?: string | null;
+  }) {
+    await this.inMemoryRateLimiter.check(
+      'inviteToOrganizationByEmail',
+      5_000, // 5 seconds
+      6, // 6 invites
+      `Exceeded rate limit for inviting to organization by email.`,
```

Comment:
Minor: I'd prefer this be in redis since it's possible for a user to be routed to a different instance. But I see this is copying the existing behavior.

Reply:
If I remember correctly this was a quick an dirty implementation for a security audit.

Reply:
I'd be in favor of creating a ticket in the backlog to improve this if we dont already have one.